code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -5,13 +5,69 @@
 
 /obj/structure/closet/crate/necropolis
 	name = "necropolis chest"
-	desc = "It's watching you closely."
+	desc = "It's watching you suspiciously."
 	icon_state = "necrocrate"
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	door_anim_time = 0
 
+/obj/structure/closet/crate/necropolis/update_icon()
+	. = ..()
+	if(locked && !opened)
+		add_overlay("necrocrate_lock")
+
+/obj/structure/closet/crate/necropolis/proc/update_desc()
+	desc = "It's watching you suspiciously."
+	if(locked)
+		desc += " It seems like it needs a key to unlock..."
+
+/obj/structure/closet/crate/necropolis/Initialize()
+	. = ..()
+	update_desc()
+
+/obj/structure/closet/crate/necropolis/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(!opened && locked)
+		// user failed to do access the chest; warning time
+		to_chat(user, "<span class='warning'>It seems that this chest is locked in a different way; you are going to need a special key for this...</span>")
+		return
+
+/obj/structure/closet/crate/necropolis/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/chest_key))
+		locked = FALSE
+		update_icon()
+		update_desc()
+		qdel(W)
+		visible_message("The [src] unlocks, destroying the key in the process!")
+
+/obj/structure/closet/crate/necropolis/emag_act(mob/user)
+	if(locked)
+		to_chat(user, "<span class='notice'>You... hit the metal lock with the card. Nothing happens. What did you expect?</span>")
+	return // No emagging a physcial lock
+
+/obj/structure/closet/crate/necropolis/bust_open()
+	if(locked)
+		for(var/obj/item/I as() in contents)
```

Comment:
```suggestion
		for(var/i in contents)
```

Reply:
You don't need an as() here because qdel isn't dependent on the type, if you wanted to be efficient you can just use a non typecasted variable.